source/builtin_macros/src/syntax.rs

```diff
@@ -164,6 +175,40 @@ impl Visitor {
                     typed.attrs.push(mk_verus_attr(token.span, quote! { proof }));
                 }
             }
+
+            // Check for Ghost(x) or Tracked(x) argument
+            use syn_verus::PatType;
+            if let FnArgKind::Typed(PatType { pat: box pat, .. }) = &mut arg.kind {
+                let mut tracked_wrapper = false;
+                let mut wrapped_pat_id = None;
+                if let Pat::TupleStruct(tup) = &pat {
+                    let ghost_wrapper = path_is_ident(&tup.path, "Ghost");
+                    tracked_wrapper = path_is_ident(&tup.path, "Tracked");
+                    if ghost_wrapper || tracked_wrapper || tup.pat.elems.len() == 1 {
+                        if let Pat::Ident(id) = &tup.pat.elems[0] {
+                            wrapped_pat_id = Some((tup.pat.elems[0].clone(), id.clone()));
+                        }
+                    }
+                }
+                if let Some((wrapped_pat, wrapped_id)) = wrapped_pat_id {
+                    use syn_verus::parse_quote_spanned;
+                    *pat = wrapped_pat;
+                    let x = wrapped_id.ident;
+                    let span = pat.span();
+                    view_ghost_tracked.push(stmt_with_semi!(span => let #x = #x.view()));
+                    // Note: we let subsequent processing desugar unwrap_ghost_tracked
+                    if is_trait {
+                        // do nothing
+                    } else if tracked_wrapper {
+                        let stmt = parse_quote_spanned!(span => let Tracked(#x) = #x;);
```

Comment:
For the record, there's a pattern feature on arguments now: `fn foo(Constructor { a }: Type)`, however we still need custom desugaring for this, so I don't think it's applicable here.